tools/build_defs/framework.bzl

```diff
@@ -377,19 +380,40 @@ def cc_external_rule_impl(ctx, attrs):
     if "requires-network" in ctx.attr.tags:
         execution_requirements = {"requires-network": ""}
 
-    ctx.actions.run_shell(
+    # We need to create a native batch script on windows to ensure the wrapper
+    # script can correctly be envoked with bash.
+    wrapper = wrapped_outputs.wrapper_script_file
+    extra_tools = []
+    if "win" in execution_os_name:
+        batch_wrapper = ctx.actions.declare_file(wrapper.short_path + ".bat")
+        ctx.actions.write(
+            output = batch_wrapper,
+            content = "\n".join([
+                "@ECHO OFF",
+                "START /b /wait bash {}".format(wrapper.path),
+                "",
+            ]),
+            is_executable = True,
+        )
+        extra_tools.append(wrapper)
+        wrapper = batch_wrapper
+
+    ctx.actions.run(
         mnemonic = "Cc" + attrs.configure_name.capitalize() + "MakeRule",
         inputs = depset(inputs.declared_inputs, transitive = [cc_toolchain.all_files]),
         outputs = rule_outputs + [
             empty.file,
             wrapped_outputs.log_file,
         ],
-        tools = depset([wrapped_outputs.script_file] + ctx.files.data + ctx.files.tools_deps + ctx.files.additional_tools, transitive = [data[DefaultInfo].default_runfiles.files for data in data_dependencies]),
+        tools = depset(
+            [wrapped_outputs.script_file] + extra_tools + ctx.files.data + ctx.files.tools_deps + ctx.files.additional_tools,
+            transitive = [data[DefaultInfo].default_runfiles.files for data in data_dependencies],
+        ),
         # We should take the default PATH passed by Bazel, not that from cc_toolchain
         # for Windows, because the PATH under msys2 is different and that is which we need
         # for shell commands
         use_default_shell_env = execution_os_name != "osx",
```

Comment:
nit: Can we update this comment to be consistent with the behaviour of what is actually happening here?

Reply:
Updated this to a `TODO` we should ideally not need this (or make it optional for users).